staging/src/github.com/kubeedge/beehive/pkg/core/context/context_channel.go

```diff
@@ -33,16 +38,18 @@ type ChannelContext struct {
 }
 
 // NewChannelContext creates and returns object of new channel context
-// TODO: Singleton
 func NewChannelContext() *ChannelContext {
-	channelMap := make(map[string]chan model.Message)
-	moduleChannels := make(map[string]map[string]chan model.Message)
-	anonChannels := make(map[string]chan model.Message)
-	return &ChannelContext{
-		channels:     channelMap,
-		typeChannels: moduleChannels,
-		anonChannels: anonChannels,
-	}
+	onceForChannel.Do(func() {
```

Comment:
I noticed several call points for this func(NewChannelContext), and I'm afraid that this function will be used to return different `ChannelContext`. 
If we use the `once`, return the same channelContext, thoughts? @sids-b @kadisi